ext/ruby.c

```diff
@@ -4,3 +4,7 @@
 VALUE RS_Qtrue = Qtrue;
 VALUE RS_Qfalse = Qfalse;
 VALUE RS_Qnil = Qnil;
+
+// NOTE: These may be less permissve than the real macros
+long RS_RSTRING_LEN(VALUE str) { return RSTRING_LEN(str); }
+uintptr_t RS_RSTRING_PTR(VALUE str) { return RSTRING_PTR(str); }
```

Comment:
this returns a (`const`?) `char *`

Reply:
Technically it's a `uintptr_t` in the Ruby C code? Should we say that we know better? (Not a rhetorical question.)

Reply:
Is it? https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L961 https://github.com/ruby/ruby/blob/v2_5_1/include/ruby/ruby.h#L967

Reply:
Hmm... maybe I'm wrong. Wonder why I got the `uintptr_t` idea from?

Reply:
How do I know whether or not to use `const`?

Reply:
I guess we should do `const` as I don't see a good case for people mutating these strings, though I guess it could work?